v3: Mixing streaming and non-streaming services generates incorrect example code

When using streaming and non-streaming services in the design, the generated server examples are incorrect.

Simple design:
```
package design

import . "goa.design/goa/v3/dsl"

var _ = API("api", func() {
	Server("apisvr", func() {
		Host("default", func() {
			URI("http://0.0.0.0:8000")
			URI("grpc://0.0.0.0:8080")
		})
	})
})

var _ = Service("ws", func() {
	HTTP(func() {
		Path("/")
	})

	Method("Retrieve", func() {
		Payload(func() {
			Field(1, "id", String, "Payload required to avoid issue #2242")
		})

		StreamingResult(String)

		HTTP(func() {
			GET("/{id}")
			Response(StatusOK)
		})

		GRPC(func() {
			Response(CodeOK)
		})
	})
})

var _ = Service("svc", func() {
	Method("test", func() {
		Result(String)

		HTTP(func() {
			GET("/")
			Response(StatusOK)
		})

		GRPC(func() {
			Response(CodeOK)
		})
	})
})
```
This generates the following endpoints in `http.go`:
```
// Wrap the endpoints with the transport specific layers. The generated
// server packages contains code generated from the design which maps
// the service input and output data structures to HTTP requests and
// responses.
var (
	wsServer  *wssvr.Server
	svcServer *svcsvr.Server
)
{
	eh := errorHandler(logger)
	upgrader := &websocket.Upgrader{}
	wsServer = wssvr.New(wsEndpoints, mux, dec, enc, eh, upgrader, nil)
	svcServer = svcsvr.New(svcEndpoints, mux, dec, enc, eh, upgrader, nil)
}
```
leading to following error:
```
cmd/apisvr/http.go:62:25: too many arguments in call to "test3/gen/http/svc/server".New
        have (*svc.Endpoints, "goa.design/goa/v3/http".Muxer, func(*"net/http".Request) "goa.design/goa/v3/http".Decoder, func(context.Context, "net/http".ResponseWriter) "goa.design/goa/v3/http".Encoder, func(context.Context, "net/http".ResponseWriter, error), *websocket.Upgrader, nil)
        want (*svc.Endpoints, "goa.design/goa/v3/http".Muxer, func(*"net/http".Request) "goa.design/goa/v3/http".Decoder, func(context.Context, "net/http".ResponseWriter) "goa.design/goa/v3/http".Encoder, func(context.Context, "net/http".ResponseWriter, error))
```
`svcsvr.New()` receives the same parameters as the streaming endpoint, but should be `svcsvr.New(svcEndpoints, mux, dec, enc, eh)`.

--- a/http/codegen/client.go
+++ b/http/codegen/client.go
@@ -50,10 +50,10 @@ func client(genpkg string, svc *expr.HTTPServiceExpr) *codegen.File {
 		Source: clientStructT,
 		Data:   data,
 		FuncMap: map[string]interface{}{
-			"streamingEndpointExists": streamingEndpointExists,
+			"hasStreaming": hasStreaming,
 		},
 	})
-	if streamingEndpointExists(data) {
+	if hasStreaming(data) {
 		sections = append(sections, &codegen.SectionTemplate{
 			Name:   "client-stream-conn-configurer-struct",
 			Source: streamConnConfigurerStructT,
@@ -88,11 +88,11 @@ func client(genpkg string, svc *expr.HTTPServiceExpr) *codegen.File {
 		Source: clientInitT,
 		Data:   data,
 		FuncMap: map[string]interface{}{
-			"streamingEndpointExists": streamingEndpointExists,
+			"hasStreaming": hasStreaming,
 		},
 	})
 
-	if streamingEndpointExists(data) {
+	if hasStreaming(data) {
 		sections = append(sections, &codegen.SectionTemplate{
 			Name:   "client-stream-conn-configurer-struct-init",
 			Source: streamConnConfigurerStructInitT,
@@ -282,7 +282,7 @@ type {{ .ClientStruct }} struct {
 	host       string
 	encoder    func(*http.Request) goahttp.Encoder
 	decoder    func(*http.Response) goahttp.Decoder
-	{{- if streamingEndpointExists . }}
+	{{- if hasStreaming . }}
 	dialer goahttp.Dialer
 	configurer *ConnConfigurer
 	{{- end }}
@@ -298,12 +298,12 @@ func New{{ .ClientStruct }}(
 	enc func(*http.Request) goahttp.Encoder,
 	dec func(*http.Response) goahttp.Decoder,
 	restoreBody bool,
-	{{- if streamingEndpointExists . }}
+	{{- if hasStreaming . }}
 	dialer goahttp.Dialer,
 	cfn *ConnConfigurer,
 	{{- end }}
 ) *{{ .ClientStruct }} {
-{{- if streamingEndpointExists . }}
+{{- if hasStreaming . }}
 	if cfn == nil {
 		cfn = &ConnConfigurer{}
 	}
@@ -317,7 +317,7 @@ func New{{ .ClientStruct }}(
 		host:              host,
 		decoder:           dec,
 		encoder:           enc,
-		{{- if streamingEndpointExists . }}
+		{{- if hasStreaming . }}
 		dialer: dialer,
 		configurer: cfn,
 		{{- end }}

--- a/http/codegen/client_cli.go
+++ b/http/codegen/client_cli.go
@@ -41,7 +41,7 @@ func ClientCLIFiles(genpkg string, root *expr.RootExpr) []*codegen.File {
 		if len(sd.Endpoints) > 0 {
 			command := &commandData{
 				CommandData: cli.BuildCommandData(sd.Service),
-				NeedStream:  streamingEndpointExists(sd),
+				NeedStream:  hasStreaming(sd),
 			}
 
 			for _, e := range sd.Endpoints {

--- a/http/codegen/example_cli.go
+++ b/http/codegen/example_cli.go
@@ -89,7 +89,8 @@ func exampleCLI(genpkg string, root *expr.RootExpr, svr *expr.ServerExpr) *codeg
 				"APIPkg":   apiPkg,
 			},
 			FuncMap: map[string]interface{}{
-				"needStream": needStream,
+				"needStream":   needStream,
+				"hasStreaming": hasStreaming,
 			},
 		},
 		&codegen.SectionTemplate{Name: "cli-http-usage", Source: httpCLIUsageT},
@@ -135,8 +136,10 @@ const (
 		debug,
 		{{- if needStream .Services }}
 		dialer,
-			{{- range .Services }}
+			{{- range $svc := .Services }}
+				{{- if hasStreaming $svc }}
 				nil,
+				{{- end }}
 			{{- end }}
 		{{- end }}
 		{{- range .Services }}

--- a/http/codegen/example_server.go
+++ b/http/codegen/example_server.go
@@ -100,7 +100,7 @@ func exampleServer(genpkg string, root *expr.RootExpr, svr *expr.ServerExpr) *co
 				"Services": svcdata,
 				"APIPkg":   apiPkg,
 			},
-			FuncMap: map[string]interface{}{"needStream": needStream},
+			FuncMap: map[string]interface{}{"needStream": needStream, "hasStreaming": hasStreaming},
 		},
 		&codegen.SectionTemplate{Name: "server-http-middleware", Source: httpSvrMiddlewareT},
 		&codegen.SectionTemplate{
@@ -249,9 +249,9 @@ func handleHTTPServer(ctx context.Context, u *url.URL{{ range $.Services }}{{ if
 	{{- if needStream .Services }}
 		upgrader := &websocket.Upgrader{}
 	{{- end }}
-	{{- range .Services }}
+	{{- range $svc := .Services }}
 		{{-  if .Endpoints }}
-		{{ .Service.VarName }}Server = {{ .Service.PkgName }}svr.New({{ .Service.VarName }}Endpoints, mux, dec, enc, eh{{ if needStream $.Services }}, upgrader, nil{{ end }}{{ range .Endpoints }}{{ if .MultipartRequestDecoder }}, {{ $.APIPkg }}.{{ .MultipartRequestDecoder.FuncName }}{{ end }}{{ end }})
+		{{ .Service.VarName }}Server = {{ .Service.PkgName }}svr.New({{ .Service.VarName }}Endpoints, mux, dec, enc, eh{{ if hasStreaming $svc }}, upgrader, nil{{ end }}{{ range .Endpoints }}{{ if .MultipartRequestDecoder }}, {{ $.APIPkg }}.{{ .MultipartRequestDecoder.FuncName }}{{ end }}{{ end }})
 		{{-  else }}
 		{{ .Service.VarName }}Server = {{ .Service.PkgName }}svr.New(nil, mux, dec, enc, eh)
 		{{-  end }}

--- a/http/codegen/server.go
+++ b/http/codegen/server.go
@@ -30,10 +30,10 @@ func serverFile(genpkg string, svc *expr.HTTPServiceExpr) *codegen.File {
 	path := filepath.Join(codegen.Gendir, "http", svcName, "server", "server.go")
 	title := fmt.Sprintf("%s HTTP server", svc.Name())
 	funcs := map[string]interface{}{
-		"join":                    func(ss []string, s string) string { return strings.Join(ss, s) },
-		"streamingEndpointExists": streamingEndpointExists,
-		"upgradeParams":           upgradeParams,
-		"viewedServerBody":        viewedServerBody,
+		"join":             func(ss []string, s string) string { return strings.Join(ss, s) },
+		"hasStreaming":     hasStreaming,
+		"upgradeParams":    upgradeParams,
+		"viewedServerBody": viewedServerBody,
 	}
 	sections := []*codegen.SectionTemplate{
 		codegen.Header(title, "server", []*codegen.ImportSpec{
@@ -58,7 +58,7 @@ func serverFile(genpkg string, svc *expr.HTTPServiceExpr) *codegen.File {
 	sections = append(sections, &codegen.SectionTemplate{Name: "server-mountpoint", Source: mountPointStructT, Data: data})
 
 	// public types
-	if streamingEndpointExists(data) {
+	if hasStreaming(data) {
 		sections = append(sections, &codegen.SectionTemplate{
 			Name:   "server-stream-conn-configurer-struct",
 			Source: streamConnConfigurerStructT,
@@ -90,7 +90,7 @@ func serverFile(genpkg string, svc *expr.HTTPServiceExpr) *codegen.File {
 	}
 
 	sections = append(sections, &codegen.SectionTemplate{Name: "server-init", Source: serverInitT, Data: data, FuncMap: funcs})
-	if streamingEndpointExists(data) {
+	if hasStreaming(data) {
 		sections = append(sections, &codegen.SectionTemplate{
 			Name:   "server-stream-conn-configurer-struct-init",
 			Source: streamConnConfigurerStructInitT,
@@ -315,7 +315,7 @@ func {{ .ServerInit }}(
 	dec func(*http.Request) goahttp.Decoder,
 	enc func(context.Context, http.ResponseWriter) goahttp.Encoder,
 	eh func(context.Context, http.ResponseWriter, error),
-	{{- if streamingEndpointExists . }}
+	{{- if hasStreaming . }}
 	up goahttp.Upgrader,
 	cfn *ConnConfigurer,
 	{{- end }}
@@ -325,7 +325,7 @@ func {{ .ServerInit }}(
 		{{- end }}
 	{{- end }}
 ) *{{ .ServerStruct }} {
-{{- if streamingEndpointExists . }}
+{{- if hasStreaming . }}
 	if cfn == nil {
 		cfn = &ConnConfigurer{}
 	}
@@ -358,7 +358,7 @@ func (s *{{ .ServerStruct }}) {{ .ServerService }}() string { return "{{ .Servic
 
 // input: ServiceData
 const serverUseT = `{{ printf "Use wraps the server handlers with the given middleware." | comment }}
-func (s *{{ .ServerStruct }}) Use(m func(http.Handler) http.Handler) { 
+func (s *{{ .ServerStruct }}) Use(m func(http.Handler) http.Handler) {
 {{- range .Endpoints }}
 	s.{{ .Method.VarName }} = m(s.{{ .Method.VarName }})
 {{- end }}
@@ -1217,7 +1217,7 @@ const responseT = `{{ define "response" -}}
 		{{ .VarName }}Slice := make([]string, len({{ .Target }}))
 		for i, e := range {{ .Target }}  {
 			{{ template "header_conversion" (headerConversionData .Type.ElemType.Type "es" true "e") }}
-			{{ .VarName }}Slice[i] = es	
+			{{ .VarName }}Slice[i] = es
 		}
 		{{ .VarName }} := strings.Join({{ .VarName }}Slice, ", ")
 		{{- end }}

--- a/http/codegen/service_data.go
+++ b/http/codegen/service_data.go
@@ -2524,16 +2524,16 @@ func upgradeParams(e *EndpointData, fn string) map[string]interface{} {
 // uses stream for sending payload/result.
 func needStream(data []*ServiceData) bool {
 	for _, svc := range data {
-		if streamingEndpointExists(svc) {
+		if hasStreaming(svc) {
 			return true
 		}
 	}
 	return false
 }
 
-// streamingEndpointExists returns true if at least one of the endpoints in
+// hasStreaming returns true if at least one of the endpoints in
 // the service defines a streaming payload or result.
-func streamingEndpointExists(sd *ServiceData) bool {
+func hasStreaming(sd *ServiceData) bool {
 	for _, e := range sd.Endpoints {
 		if isStreamingEndpoint(e) {
 			return true

